pkg/services/live/features/watch.go

```diff
@@ -0,0 +1,200 @@
+	if ok && !current.done {
+		return model.SubscribeReply{
+			JoinLeave: false,
+			Presence:  false,
+			Recover:   false,
+		}, backend.SubscribeStreamStatusOK, nil
+	}
+
+	// Try to start a watcher for this request
+	gvr, name, err := parseWatchRequest(e.Channel, userID)
+	if err != nil {
+		return model.SubscribeReply{}, backend.SubscribeStreamStatusNotFound, err
+	}
+
+	// Test this with only provisiong support -- then we can evaluate a broader rollout
+	if gvr.Group != provisioning.GROUP {
+		return model.SubscribeReply{}, backend.SubscribeStreamStatusPermissionDenied,
+			fmt.Errorf("watching provisioned resources is OK allowed (for now)")
+	}
+
+	requester := types.WithAuthInfo(context.Background(), u)
+	cfg, err := b.configProvider.GetRestConfig(requester)
+	if err != nil {
+		return model.SubscribeReply{}, backend.SubscribeStreamStatusNotFound, err
+	}
+	uclient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return model.SubscribeReply{}, backend.SubscribeStreamStatusNotFound, err
+	}
+	client := uclient.Resource(gvr).Namespace(u.GetNamespace())
+
+	opts := v1.ListOptions{}
+	if len(name) > 1 {
+		opts.FieldSelector = "metadata.name=" + name
+	}
+	watch, err := client.Watch(requester, opts)
+	if err != nil {
+		return model.SubscribeReply{}, backend.SubscribeStreamStatusNotFound, err
+	}
+
+	current = &watcher{
+		orgId:     u.GetOrgID(),
+		channel:   e.Channel,
+		publisher: b.publisher,
+		watch:     watch,
+	}
+
+	b.watching[e.Channel] = current
+	go current.run(ctx)
+
+	return model.SubscribeReply{
+		JoinLeave: false, // need unsubscribe envents
+		Presence:  false,
+		Recover:   false,
+	}, backend.SubscribeStreamStatusOK, nil
+}
+
+func parseWatchRequest(channel string, user string) (gvr schema.GroupVersionResource, name string, err error) {
+	addr, err := live.ParseChannel(channel)
+	if err != nil {
+		return gvr, "", err
+	}
+
+	parts := strings.Split(addr.Path, "/")
+	if len(parts) != 3 {
+		return gvr, "", fmt.Errorf("expecting path: {version}/{resource}={name}/{user}")
+	}
+	if parts[2] != user {
+		return gvr, "", fmt.Errorf("expecting user suffix: %s", user)
+	}
+
+	resource := strings.Split(parts[1], "=")
+	gvr = schema.GroupVersionResource{
+		Group:    addr.Namespace,
+		Version:  parts[0],
+		Resource: resource[0],
+	}
+	if len(resource) > 1 {
+		name = resource[1]
+	}
+	return gvr, name, nil
+}
+
+// OnPublish is called when a client wants to broadcast on the websocket
+func (b *WatchRunner) OnPublish(_ context.Context, u identity.Requester, e model.PublishEvent) (model.PublishReply, backend.PublishStreamStatus, error) {
+	return model.PublishReply{}, backend.PublishStreamStatusNotFound, fmt.Errorf("watch does not support publish")
+}
+
+type watcher struct {
+	orgId     int64
+	channel   string
+	publisher model.ChannelPublisher
+	done      bool
+	watch     watch.Interface
+}
+
+func (b *watcher) run(ctx context.Context) {
+	logger := logging.FromContext(ctx).With("channel", b.channel)
```

Comment:
did you intend to use the app SDK logger here?

Reply:
yes -- I *think* this is what we should use moving forward (but honestly I have no strong opinions here)